overlord/configstate/config/helpers.go

```diff
@@ -88,6 +91,17 @@ func PatchConfig(snapName string, subkeys []string, pos int, config interface{},
 // The provided key may be formed as a dotted key path through nested maps.
 // For example, the "a.b.c" key describes the {a: {b: {c: value}}} map.
 func GetFromChange(snapName string, subkeys []string, pos int, config map[string]interface{}, result interface{}) error {
+	// special case - get root document
+	if len(subkeys) == 0 {
+		if config == nil {
+			return &NoOptionError{SnapName: snapName, Key: ""}
```

Comment:
Good point, it would result in "snap 'foo' has no '' configuration option. I've now special-cased it for root document (the logic is in getFromPristine to say "snap "foo" has no configuration".